### PR TITLE
feat(UNT-T10112): implement theme switch component

### PR DESCRIPTION
**Pull request by chandani-c-simform**
- Implement theme switch component

### CODE REVIEW

File: src/components/AnimatedSwitch/AnimatedSwitch.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  darkThemeColor = Colors.darkModerateViolet,
+}: Partial<AnimatedSwitchProps>) => {
+  const {
+    toggle,
+    interpolateColorBackGround,
+    overrideNotchAnimatedStyle,
+    notchAnimatedStyle,
+  } = useAnimatedSwitch({ size, lightThemeColor, darkThemeColor });
+
+  const style = styles({
```
Comment (shalin-simform) [inline src/components/AnimatedSwitch/AnimatedSwitch.tsx:21]: use destructuring

File: src/components/AnimatedSwitch/AnimatedSwitch.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    size,
+  });
+
+  return (
+    <>
```
Comment (shalin-simform) [inline src/components/AnimatedSwitch/AnimatedSwitch.tsx:26]: remove this fragment

File: src/components/AnimatedSwitch/hooks/useAnimatedSwitch.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  lightThemeColor,
+  darkThemeColor,
+}: AnimatedSwitchProps) => {
+  const [darkTheme, setDarkTheme] = useState(false);
+  const offSet = useSharedValue(0);
+  const overSet = useSharedValue(0);
+
+  const toggle = useCallback(() => {
+    if (offSet.value === 0) {
+      offSet.value = withTiming(0.5, { duration: 810 });
```
Comment (shalin-simform) [inline src/components/AnimatedSwitch/hooks/useAnimatedSwitch.tsx:21]: use optional chaining

File: src/components/AnimatedSwitch/hooks/useAnimatedSwitch.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  const overrideNotchAnimatedStyle = useAnimatedStyle(() => {
+    return {
+      transform: [
+        { translateX: overSet.value * size * 4 },
+        { translateY: overSet.value * size * -0.18 },
+      ],
+    };
+  }, []);
+
+  const interpolateColorBackGround = useAnimatedStyle(() => {
```
Comment (shalin-simform) [inline src/components/AnimatedSwitch/hooks/useAnimatedSwitch.tsx:50]: update variable name

File: src/components/AnimatedSwitch/styles.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { PixelRatio, StyleSheet } from 'react-native';
+import { Colors } from '../../theme';
+import type { StyleProps } from './types';
+
+const styles = ({ size }: StyleProps) =>
+  StyleSheet.create({
+    buttonContainer: {
+      height: PixelRatio.roundToNearestPixel(size),
```
Comment (shalin-simform) [inline src/components/AnimatedSwitch/styles.ts:8]: move PixelRatio logic into Metrics file

File: src/components/AnimatedSwitch/index.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+import AnimatedSwitch from './AnimatedSwitch';
+
+export { AnimatedSwitch };
```
Comment (shalin-simform) [inline src/components/AnimatedSwitch/index.ts:3]: export types and hooks from index file

File: src/components/AnimatedSwitch/AnimatedSwitch.tsx (hunk 2 above)
Comment (chandani-c-simform) [reply src/components/AnimatedSwitch/AnimatedSwitch.tsx:26]: resolved

File: src/components/AnimatedSwitch/hooks/useAnimatedSwitch.tsx (hunk 4 above)
Comment (chandani-c-simform) [reply src/components/AnimatedSwitch/hooks/useAnimatedSwitch.tsx:50]: resolved

File: src/components/AnimatedSwitch/AnimatedSwitch.tsx (hunk 1 above)
Comment (chandani-c-simform) [reply src/components/AnimatedSwitch/AnimatedSwitch.tsx:21]: resolved

File: src/components/AnimatedSwitch/index.ts (hunk 6 above)
Comment (chandani-c-simform) [reply src/components/AnimatedSwitch/index.ts:3]: resolved

File: src/components/AnimatedSwitch/styles.ts (hunk 5 above)
Comment (chandani-c-simform) [reply src/components/AnimatedSwitch/styles.ts:8]: resolved

File: src/components/AnimatedSwitch/hooks/useAnimatedSwitch.tsx (hunk 3 above)
Comment (chandani-c-simform) [reply src/components/AnimatedSwitch/hooks/useAnimatedSwitch.tsx:21]: resolved

File: example/.prettierrc.js [hunk 7]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  bracketSpacing: false,
```
Comment (shalin-simform) [inline example/.prettierrc.js:2]: remove this changes

File: example/src/modules/AnimatedSwitch/styles/styles.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import {StyleSheet} from 'react-native';
+import {Colors, Metrics} from '../../../theme';
+
+const {moderateScale, verticalScale} = Metrics;
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+  },
+  firstVariantContainer: {
```
Comment (shalin-simform) [inline example/src/modules/AnimatedSwitch/styles/styles.ts:10]: give proper name to style object

File: example/src/modules/AnimatedSwitch/styles/styles.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    padding: moderateScale(20),
+  },
+  firstVariantTitle: {
```
Comment (shalin-simform) [inline example/src/modules/AnimatedSwitch/styles/styles.ts:13]: give a proper name to the style object

File: example/src/modules/AnimatedSwitch/styles/styles.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+    fontSize: moderateScale(16),
+    fontWeight: '500',
+    letterSpacing: 1,
+    color: Colors.black,
+  },
+  firstSwitchContainer: {
```
Comment (shalin-simform) [inline example/src/modules/AnimatedSwitch/styles/styles.ts:19]: give a proper name to the style object

File: example/.prettierrc.js (hunk 7 above)
Comment (chandani-c-simform) [reply example/.prettierrc.js:2]: resolved

File: example/src/modules/AnimatedSwitch/styles/styles.ts (hunk 8 above)
Comment (chandani-c-simform) [reply example/src/modules/AnimatedSwitch/styles/styles.ts:10]: resolved

File: example/src/modules/AnimatedSwitch/styles/styles.ts (hunk 9 above)
Comment (chandani-c-simform) [reply example/src/modules/AnimatedSwitch/styles/styles.ts:13]: resolved

File: example/src/modules/AnimatedSwitch/styles/styles.ts (hunk 10 above)
Comment (chandani-c-simform) [reply example/src/modules/AnimatedSwitch/styles/styles.ts:19]: resolved

**Review by shalin-simform — APPROVED** [review]

**Pull request merged**